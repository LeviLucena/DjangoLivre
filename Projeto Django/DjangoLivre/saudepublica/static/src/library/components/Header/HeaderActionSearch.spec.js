import { shallowMount } from '@vue/test-utils'
import BrButton from '../Button/Button.ce.vue'
import BrItem from '../Item/Item.ce.vue'
import BrList from '../List/List.ce.vue'
import BrHeader from './Header.ce.vue'
import BrHeaderAction from './HeaderAction.ce.vue'
import BrHeaderActionSearch from './HeaderActionSearch.ce.vue'

describe('HeaderAction - HeaderActionSearch', () => {
  const slotActionsFunctions = `
  <div slot="headerAction">
    <br-header-action
    title-functions="Funcionalidades do Sistema"
    list-functions="[
      {
        icon: 'chart-bar',
        name: 'Funcionalidade 1',
        url: '#'
      }
    ]"
    title-links="Acesso RÃ¡pido"
    list-links="[
    {
      name: 'Link de acesso 1',
      href: '#'
    },
    {
      name: 'Link de acesso 2',
      href: '#'
    },
    {
      name: 'Link de acesso 3',
      href: '#'
    }
  ]"
    />
  </div>
  `

  test('it emits "open-search" and calls "openSearch" method', async () => {
    const spyOpenSearch = jest.spyOn(BrHeaderActionSearch.methods, 'openSearch')
    const wrapper = shallowMount(BrHeader, {
      props: {
        hasSearch: true,
        compact: true,
      },
      slots: {
        headerAction: slotActionsFunctions,
      },
      global: {
        stubs: {
          'br-header-action': BrHeaderAction,
          'br-header-action-search': BrHeaderActionSearch,
          'br-list': BrList,
          'br-item': BrItem,
          'br-button': BrButton,
        },
      },
    })
    const childWrapper = wrapper.findComponent(BrHeaderActionSearch)
    childWrapper.vm.openSearch()
    childWrapper.vm.$emit('open-search')
    expect(childWrapper.emitted('open-search')).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
    wrapper.find('[data-target=".header-search"]').trigger('click')
    await wrapper.vm.$nextTick()
    expect(spyOpenSearch).toHaveBeenCalled()
    expect(wrapper.find('.header-links.dropdown.show').exists()).toBe(false)
    expect(wrapper.find('.header-functions.dropdown.show').exists()).toBe(false)
    expect(wrapper.find('.header-search.active').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })
})
