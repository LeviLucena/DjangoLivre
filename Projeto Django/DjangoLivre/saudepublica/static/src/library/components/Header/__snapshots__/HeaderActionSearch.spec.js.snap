// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderAction - HeaderActionSearch it emits "open-search" and calls "openSearch" method 1`] = `
<header
  class="br-header mb-4 compact"
>
  <div
    class="container-lg"
  >
    <div
      class="header-top"
    >
      <div
        class="header-logo"
      >
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      
      <div
        slot="headerAction"
      >
        <div
          class="header-actions"
          style="float: right;"
        >
          <div
            class="header-links dropdown"
          >
            <button
              aria-label="Acesso Rápido"
              class="br-button small circle"
              data-toggle="dropdown"
              type="button"
            >
              
              <icon-base-stub
                class=""
                familyname="fas"
                iconname="ellipsis-v"
              />
              
            </button>
            <div
              class="br-list"
              data-toggle="false"
              data-unique="false"
              density="large"
              role="list"
            >
              <div
                class="header"
              >
                <div
                  class="title"
                >
                  Acesso Rápido
                </div>
              </div>
              <!--v-if-->
              
              
              <div
                class="item-wrapper"
                name="Link de acesso 1"
              >
                <a
                  class="br-item"
                  data-toggle="selection"
                  href="#"
                  name="Link de acesso 1"
                >
                  <div
                    class="content"
                  >
                    
                    Link de acesso 1
                    
                  </div>
                </a>
                <span
                  style="display: none;"
                >
                  
                  Link de acesso 1
                  
                </span>
              </div>
              <div
                class="item-wrapper"
                name="Link de acesso 2"
              >
                <a
                  class="br-item"
                  data-toggle="selection"
                  href="#"
                  name="Link de acesso 2"
                >
                  <div
                    class="content"
                  >
                    
                    Link de acesso 2
                    
                  </div>
                </a>
                <span
                  style="display: none;"
                >
                  
                  Link de acesso 2
                  
                </span>
              </div>
              <div
                class="item-wrapper"
                name="Link de acesso 3"
              >
                <a
                  class="br-item"
                  data-toggle="selection"
                  href="#"
                  name="Link de acesso 3"
                >
                  <div
                    class="content"
                  >
                    
                    Link de acesso 3
                    
                  </div>
                </a>
                <span
                  style="display: none;"
                >
                  
                  Link de acesso 3
                  
                </span>
              </div>
              
              
            </div>
          </div>
          <br-divider-stub
            class="mx-half mx-sm-1"
            dashed="false"
            vertical="true"
          />
          <div
            class="header-functions dropdown"
          >
            <button
              aria-label="Funcionalidades do Sistema"
              class="br-button small circle"
              data-toggle="dropdown"
              type="button"
            >
              
              <icon-base-stub
                class=""
                familyname="fas"
                iconname="th"
              />
              
            </button>
            <div
              class="br-list"
              data-toggle="false"
              data-unique="false"
              density="large"
              role="list"
            >
              <div
                class="header"
              >
                <div
                  class="title"
                >
                  Funcionalidades do Sistema
                </div>
              </div>
              <!--v-if-->
              
              
              <div
                class="item-wrapper align-items-center"
              >
                <a
                  class="align-items-center br-item"
                  href="#"
                >
                  <div
                    class="content"
                  >
                    
                    <button
                      aria-label=""
                      class="br-button small circle"
                      title="Funcionalidade 1"
                      type="button"
                    >
                      
                      <icon-base-stub
                        class=""
                        familyname="fas"
                        iconname="chart-bar"
                      />
                      <span
                        class="text"
                      >
                        Funcionalidade 1
                      </span>
                      
                    </button>
                    
                  </div>
                </a>
                <span
                  style="display: none;"
                >
                  
                  <button
                    aria-label=""
                    class="br-button small circle"
                    title="Funcionalidade 1"
                    type="button"
                  >
                    
                    <icon-base-stub
                      class=""
                      familyname="fas"
                      iconname="chart-bar"
                    />
                    <span
                      class="text"
                    >
                      Funcionalidade 1
                    </span>
                    
                  </button>
                  
                </span>
              </div>
              
              
            </div>
          </div>
          <div
            class="header-search-trigger"
            style="display: block;"
          >
            <button
              aria-label="Abrir Busca"
              class="br-button middle circle"
              data-target=".header-search"
              data-toggle="search"
              type="button"
            >
              
              <icon-base-stub
                class=""
                familyname="fas"
                iconname="search"
              />
              <!--v-if-->
              
            </button>
          </div>
          <!--v-if-->
        </div>
      </div>
      
    </div>
    <div
      class="header-bottom"
    >
      <div
        class="header-menu"
      >
        <!--v-if-->
        <div
          class="header-info"
        >
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div
        class="header-search"
      >
        <br-header-search-stub
          id="searchbox1"
          label="Texto da pesquisa"
          placeholder="O que você procura?"
        />
      </div>
    </div>
  </div>
</header>
`;

exports[`HeaderAction - HeaderActionSearch it emits "open-search" and calls "openSearch" method 2`] = `
<header
  class="br-header mb-4 compact"
>
  <div
    class="container-lg"
  >
    <div
      class="header-top"
    >
      <div
        class="header-logo"
      >
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      
      <div
        slot="headerAction"
      >
        <div
          class="header-actions"
          style="float: right;"
        >
          <div
            class="header-links dropdown"
          >
            <button
              aria-label="Acesso Rápido"
              class="br-button small circle"
              data-toggle="dropdown"
              type="button"
            >
              
              <icon-base-stub
                class=""
                familyname="fas"
                iconname="ellipsis-v"
              />
              
            </button>
            <div
              class="br-list"
              data-toggle="false"
              data-unique="false"
              density="large"
              role="list"
            >
              <div
                class="header"
              >
                <div
                  class="title"
                >
                  Acesso Rápido
                </div>
              </div>
              <!--v-if-->
              
              
              <div
                class="item-wrapper"
                name="Link de acesso 1"
              >
                <a
                  class="br-item"
                  data-toggle="selection"
                  href="#"
                  name="Link de acesso 1"
                >
                  <div
                    class="content"
                  >
                    
                    Link de acesso 1
                    
                  </div>
                </a>
                <span
                  style="display: none;"
                >
                  
                  Link de acesso 1
                  
                </span>
              </div>
              <div
                class="item-wrapper"
                name="Link de acesso 2"
              >
                <a
                  class="br-item"
                  data-toggle="selection"
                  href="#"
                  name="Link de acesso 2"
                >
                  <div
                    class="content"
                  >
                    
                    Link de acesso 2
                    
                  </div>
                </a>
                <span
                  style="display: none;"
                >
                  
                  Link de acesso 2
                  
                </span>
              </div>
              <div
                class="item-wrapper"
                name="Link de acesso 3"
              >
                <a
                  class="br-item"
                  data-toggle="selection"
                  href="#"
                  name="Link de acesso 3"
                >
                  <div
                    class="content"
                  >
                    
                    Link de acesso 3
                    
                  </div>
                </a>
                <span
                  style="display: none;"
                >
                  
                  Link de acesso 3
                  
                </span>
              </div>
              
              
            </div>
          </div>
          <br-divider-stub
            class="mx-half mx-sm-1"
            dashed="false"
            vertical="true"
          />
          <div
            class="header-functions dropdown"
          >
            <button
              aria-label="Funcionalidades do Sistema"
              class="br-button small circle"
              data-toggle="dropdown"
              type="button"
            >
              
              <icon-base-stub
                class=""
                familyname="fas"
                iconname="th"
              />
              
            </button>
            <div
              class="br-list"
              data-toggle="false"
              data-unique="false"
              density="large"
              role="list"
            >
              <div
                class="header"
              >
                <div
                  class="title"
                >
                  Funcionalidades do Sistema
                </div>
              </div>
              <!--v-if-->
              
              
              <div
                class="item-wrapper align-items-center"
              >
                <a
                  class="align-items-center br-item"
                  href="#"
                >
                  <div
                    class="content"
                  >
                    
                    <button
                      aria-label=""
                      class="br-button small circle"
                      title="Funcionalidade 1"
                      type="button"
                    >
                      
                      <icon-base-stub
                        class=""
                        familyname="fas"
                        iconname="chart-bar"
                      />
                      <span
                        class="text"
                      >
                        Funcionalidade 1
                      </span>
                      
                    </button>
                    
                  </div>
                </a>
                <span
                  style="display: none;"
                >
                  
                  <button
                    aria-label=""
                    class="br-button small circle"
                    title="Funcionalidade 1"
                    type="button"
                  >
                    
                    <icon-base-stub
                      class=""
                      familyname="fas"
                      iconname="chart-bar"
                    />
                    <span
                      class="text"
                    >
                      Funcionalidade 1
                    </span>
                    
                  </button>
                  
                </span>
              </div>
              
              
            </div>
          </div>
          <div
            class="header-search-trigger"
            style="display: block;"
          >
            <button
              aria-label="Abrir Busca"
              class="br-button middle circle"
              data-target=".header-search"
              data-toggle="search"
              type="button"
            >
              
              <icon-base-stub
                class=""
                familyname="fas"
                iconname="search"
              />
              <!--v-if-->
              
            </button>
          </div>
          <!--v-if-->
        </div>
      </div>
      
    </div>
    <div
      class="header-bottom"
    >
      <div
        class="header-menu"
      >
        <!--v-if-->
        <div
          class="header-info"
        >
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div
        class="header-search active"
      >
        <br-header-search-stub
          id="searchbox1"
          label="Texto da pesquisa"
          placeholder="O que você procura?"
        />
      </div>
    </div>
  </div>
</header>
`;
